### PR TITLE
Release 10.4.2

### PR DESCRIPTION
**Pull request by fxthomas**
This is a release for fixing #1333.

We should probably also update the user guide at https://airsonic.github.io/docs/database/#mariadb (is that info still current?), but I would rather have someone familiar with MariaDB do it correctly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,12 @@
 # airsonic/airsonic
 # -->
 
+## v10.4.2 - 22 Oct 2019
+
+### Fixes
+
+- Fix a migration issue on MariaDB (#1333)
```
Comment (jvoisin) [inline CHANGELOG.md]: Since the changelog is written in markdown, maybe it's worth providing a link to the issue?

Comment (fxthomas) [reply CHANGELOG.md]: Hum, the link is added by GitHub on the releases page anyway (and is useless without GitHub). I'll do that next time? :wink:

**Comment by fxthomas** [comment]
Sure! I'll merge this and create/sign a release on GitHub for now.

I don't, however, have the keys to push a Docker image so I'll need your help on that. My key is 262121A22A609283A6A46BCB7DEEDDBFC5A13AB4.

**Comment by fxthomas** [comment]
@muff1nman Can you walk me through getting the Docker image on the registry? The GitHub release should be ready!

**Comment by muff1nman** [comment]
Sure for starters can you  open a PR to add your gpg key to the passwords repo like this one: https://github.com/airsonic/airsonic-passwords/pull/1

**Comment by muff1nman** [comment]
@fxthomas you should be able to follow the rest of the releases doc now. The image should already be built assuming you used the docker or integration-test profile when building the war

**Comment by fxthomas** [comment]
Yup, that worked. Release is up on GitHub and uploading to Docker registry!

**Comment by fxthomas** [comment]
...and done. Hopefully I did not break anything! :smile:

**Comment by muff1nman** [comment]
This doesn't look quite right to me. The 10.4.2 tag be on the release-10.4 branch:

```
$ git show-branch origin/release-10.4  origin/master  v10.4.2 
! [origin/release-10.4] Merge pull request #1362 from fxthomas/1333-allow-previous-migration-checksums-for-mariadb-release-10.4.1
 ! [origin/master] Removed unused variables, imports, empty statements (#1361)
  ! [v10.4.2] Use the try-with-resource construct where possible
---
 +  [origin/master] Removed unused variables, imports, empty statements (#1361)
 +  [origin/master^^2] Fix Test Queries for limits
 +  [origin/master~2] Cleanup extraneous quote
 +  [origin/master~3] Document a bit our review process (#1353)
 ++ [v10.4.2] Use the try-with-resource construct where possible
 ++ [v10.4.2^] Additions to gitignore for static analysis plugins
 ++ [v10.4.2~2] Remove openjdk-ea for now
 ++ [v10.4.2~3] Changed final constant to add static
 ++ [v10.4.2~4] Add OpenJDK 13 and OpenJDK-dev to travis
 ++ [v10.4.2~5] Refactor a bit how we're handling avatars
 -- [v10.4.2~6] Merge pull request #1359 from randomnicode/minorfixtostatictimeunit
 ++ [v10.4.2~6^2] Minor fix to static time unit call
 -- [v10.4.2~7] Merge pull request #1352 from fxthomas/1333-allow-previous-migration-checksums-for-mariadb
 ++ [v10.4.2~7^2] Allow previous Liquibase migration checksums for MariaDB
 ++ [v10.4.2~8] Bump apache's cxf
 ++ [v10.4.2~9] Bump OWASP's maven dependency's check
 ++ [v10.4.2~10] Use a static access in a test file, because we can.
 ++ [v10.4.2~11] Bump jackson to a proper release
 -- [v10.4.2~12] Merge pull request #1265 from benferreri/master
 ++ [v10.4.2~12^2] escaped artist and album values in links; fixed allmusic link
 -- [v10.4.2~12^2^] Merge remote-tracking branch 'upstream/master'
 ++ [v10.4.2~12^2~2] added discogs search link to main album page
 ++ [v10.4.2~13] Replace deprecated Date by Calendar
 ++ [v10.4.2~14] Minor refactor of CsrfSecurityRequestMatcher
 -- [v10.4.2~15] Merge remote-tracking branch 'origin/pr/1315'
 ++ [v10.4.2~15^2] Remove a dead file
 -- [v10.4.2~16] Merge remote-tracking branch 'origin/pr/1324'
 ++ [v10.4.2~16^2] Fix the syntax for sunfire's exclusion
 -- [v10.4.2~17] Merge remote-tracking branch 'origin/pr/1327'
 ++ [v10.4.2~17^2] Remove a homemade implementation of Pair
 -- [v10.4.2~18] Merge remote-tracking branch 'origin/pr/1335'
 ++ [v10.4.2~18^2] Fixed inconsistencies in SVG icon colors
 -- [v10.4.2~19] Merge remote-tracking branch 'origin/pr/1332'
 ++ [v10.4.2~19^2] Added test case sources to checkstyle
 -- [v10.4.2~20] Merge remote-tracking branch 'origin/pr/1330'
 ++ [v10.4.2~20^2] Fix broken update link
 -- [v10.4.2~21] Merge remote-tracking branch 'origin/pr/1313'
 ++ [v10.4.2~21^2] Added separate user icon from logout icon
 ++ [v10.4.2~22] Update changelog with 10.4.1 release
 ++ [v10.4.2~23] Bump jwt
 ++ [v10.4.2~24] Mark a field as private
 -- [v10.4.2~25] Merge remote-tracking branch 'origin/pr/1310'
 ++ [v10.4.2~25^2] Display file id and path in debug scanning logs
 -- [v10.4.2~26] Merge remote-tracking branch 'origin/pr/1312'
 ++ [v10.4.2~26^2] Remove white underline from player controls
 ++ [v10.4.2~26^2^] Update dark icons for the 'groove' theme
 -- [v10.4.2~27] Merge remote-tracking branch 'origin/pr/1302'
 ++ [v10.4.2~27^2] Add logger name checkstyle
 ++ [v10.4.2~28] Simplify a condition
 ++ [v10.4.2~29] Upgrade PlaylistService to non-deprecated methods
 ++ [v10.4.2~30] Updated the issue template to add language
 ++ [v10.4.2~31] Improved handling of missing/failed transcoder
 ++ [v10.4.2~32] Ensure commons logging is excluded (#1274)
 ++ [v10.4.2~33] Removed inability to change roles for admin user
 ++ [v10.4.2~34] Check the return value of `delete` calls
 ++ [v10.4.2~35] Replace custom circular buffer with Apache commons(CircularFifoQueue)
 ++ [v10.4.2~36] Use EnumMap in IndexManager
 ++ [v10.4.2~37] Make test results quieter
 ++ [v10.4.2~38] Remove a leftover of the Radexo removal
 ++ [v10.4.2~39] Make coverity even quieter
 ++ [v10.4.2~40] Shorten output
 ++ [v10.4.2~41] Try compile
 ++ [v10.4.2~42] Revert "Revert coverity in travis - takes too long"
 ++ [v10.4.2~43] Revert coverity in travis - takes too long
 ++ [v10.4.2~44] Try to fix travis
 ++ [v10.4.2~45] Improve Windows system detection
 ++ [v10.4.2~46] Add coverity to travis-ci
 ++ [v10.4.2~47] Remove radeox
 ++ [v10.4.2~48] Remove unnecessary exceptions and unused imports
 ++ [v10.4.2~49] Fix incorrect absolute paths in css
 ++ [v10.4.2~50] Remove Tomcat version override
 ++ [v10.4.2~51] Switch to newer modelAttribute
 -- [v10.4.2~52] Merge pull request #1279 from jvoisin/fix_javadoc
 ++ [v10.4.2~52^2] Fix some javadoc comments
 ++ [v10.4.2~53] Fix some javadoc comments
 ++ [v10.4.2~54] More lambdaification
 ++ [v10.4.2~55] Option to disable timestamps during scan
 ++ [v10.4.2~56] Media file scan will now heed configured music/video file types
 ++ [v10.4.2~57] Handle nulls when processing cover art better
 ++ [v10.4.2~58] Bump Spring
 ++ [v10.4.2~59] Please checkstyle
 ++ [v10.4.2~60] Use lambdas instead of ghetto nested types
 ++ [v10.4.2~61] Bump ehcache
 ++ [v10.4.2~62] Replace deprecated IOUtils.closeQuietly calls
 ++ [v10.4.2~63] Add expunge to IndexManager
 ++ [v10.4.2~64] Use `computeIfAbsent` instead of an `if`
 ++ [v10.4.2~65] Remove an unused import
 ++ [v10.4.2~66] Use `removeIf` instead of doing this manually
 ++ [v10.4.2~67] fixed 'Settings Saved' notification when saving transcoding settings #1114 (#1269)
 -- [v10.4.2~68] Merge branch 'origin/pr/1244'
 ++ [v10.4.2~68^2] Make the logout icon/image cliquable
 ++ [v10.4.2~69] Making textareas vertical-align middle allows them to align with adjacent imgs.
 ++ [v10.4.2~70] Fix the CSS for groove_simple
 -- [v10.4.2~71] Merge pull request #1254 from fxthomas/1160-fix-mejs-race-condition
 ++ [v10.4.2~71^2] Fix race condition in MediaElement.js (#685, #1160)
 -- [v10.4.2~72] Merge branch 'origin/pr/1248'
 ++ [v10.4.2~72^2] Upgrade jackson-* to 2.10.0.pr3 to fix CVE-2019-16335, CVE-2019-14540
 ++ [v10.4.2~73] Use a constant for encoding
 ++ [v10.4.2~74] Improve transcoder info text and formatting
 ++ [v10.4.2~75] Gracefully handle failed version check attempts
 ++ [v10.4.2~76] Change integration tests from parallel to single
 ++ [v10.4.2~77] Added checkstyle to prohibit windows line endings
 ++ [v10.4.2~78] Update Lucene from 3.0.3 to the current version 8.2.0
 ++ [v10.4.2~79] Fix to clean up docker instance on RunTimeException in Integration Tests
 ++ [v10.4.2~80] Update db-changelog.xml to support mariadb
 ++ [v10.4.2~81] Simplify a bit the annotations
 ++ [v10.4.2~82] Make the systemd unit compatible with tomcat9
 ++ [v10.4.2~83] Improve error handling for jaudiotaggerParser artwork fetch
 ++ [v10.4.2~84] Remove an unused/unuseable/half-done cache
 ++ [v10.4.2~85] Fix two issues found by LGTM
 ++ [v10.4.2~86] Ignore yet an other jackson vulnerability
 ++ [v10.4.2~87] Remove use useless arrays creation
 ++ [v10.4.2~88] Added handling for LyricsService ConnectTimeoutException
 ++ [v10.4.2~89] Handle Lyrics fetch HttpResponseException
 ++ [v10.4.2~90] upgrade jackson-databind and commons-beanutils for CVEs
 ++ [v10.4.2~91] normalize license expire date to prevent outstanding value on Java 12
 ++ [v10.4.2~92] Enable checkstyle OverloadMethodsDeclarationOrder
 ++ [v10.4.2~93] Enable checkstyle FileTabCharacter
 ++ [v10.4.2~94] Enable checkstyle UniqueProperties
 ++ [v10.4.2~95] Enable several more code checkstyles
 ++ [v10.4.2~96] Enable checkstyle for simple boolean expressions
 ++ [v10.4.2~97] Enable additional checkstyles for whitespace
 ++ [v10.4.2~98] Enable checkstyle OneStatementPerLine
 ++ [v10.4.2~99] Enable several more checkstyle modules
 ++ [v10.4.2~100] Enable checkstyle's Indentation module
 ++ [v10.4.2~101] Enable EqualsAvoidNull in checkstyle
 ++ [v10.4.2~102] Fix null exception when creating a new podcast channel
 ++ [v10.4.2~103] Ignore CVE-2019-14439 too
 ++ [v10.4.2~104] Ignore CVE-2019-14379
 ++ [v10.4.2~105] Temporarily skip a test
 ++ [v10.4.2~106] Replace a now-defunct maven repo with a working one
 ++ [v10.4.2~107] Sort the checkstyle modules
 ++ [v10.4.2~108] Add a check for InnerAssignment in checkstyle.xml
 ++ [v10.4.2~109] Remove an empty statement
 ++ [v10.4.2~110] Add more style checks
 ++ [v10.4.2~111] Enabled StringLiteralEquality in CheckStyle
 ++ [v10.4.2~112] Enable EqualsHashCode in CheckStyle
 ++ [v10.4.2~113] Synchronize on activeAudioPlayers when changing them
 ++ [v10.4.2~114] Precompile jsp
 ++ [v10.4.2~115] Fix #1139
 ++ [v10.4.2~116] Fix podcast download issues
 ++ [v10.4.2~117] Use titles in some <img> tags
 ++ [v10.4.2~118] Fixup eccb7e08f586b3149cc5a470ed225ba511f9e0d5 (#1170)
 ++ [v10.4.2~119] Fix a NULL-deref in CoverArtController.java
 ++ [v10.4.2~120] First pass of CSP-compliance for mousetrap
 ++ [v10.4.2~121] Some release note fixes
 ++ [v10.4.2~122] Bump version
-   [origin/release-10.4] Merge pull request #1362 from fxthomas/1333-allow-previous-migration-checksums-for-mariadb-release-10.4.1
+   [origin/release-10.4^2] Update changelog and bump versions to 10.4.2
+   [origin/release-10.4^2^] Allow previous Liquibase migration checksums for MariaDB
+   [origin/release-10.4^] Add date to changelog
+   [origin/release-10.4~2] Bump version to 10.4.1
+   [origin/release-10.4~3] Change integration tests from parallel to single
+   [origin/release-10.4~4] Temporarily skip a test
+   [origin/release-10.4~5] Upgrade jackson-* to 2.10.0.pr3 to fix CVE-2019-16335, CVE-2019-14540
+   [origin/release-10.4~6] upgrade jackson-databind and commons-beanutils for CVEs
+   [origin/release-10.4~7] Replace a now-defunct maven repo with a working one
+   [origin/release-10.4~8] Update db-changelog.xml to support mariadb
+   [origin/release-10.4~9] Fix race condition in MediaElement.js (#685, #1160)
+   [origin/release-10.4~10] Release 10.4.0
+++ [v10.4.2~123] Add date for changelog
```

**Comment by fxthomas** [comment]
That sounds correct to me, or at least according to what I understood of the release docs. Is there something I didn't get?

The way I see it, the `release-10.4`  branch contains the last released patch version (now 10.4.2, a maintenance/patch release) in the 10.4 series (minor release). So everything works from my point of view.

**Comment by muff1nman** [comment]
> The way I see it, the release-10.4 maintenance branch contains the last released patch version (now 10.4.2, a maintenance/patch release) in the 10.4 series (minor release). So everything works from my point of view.

Correct, however the `v10.4.2` tag is on the master branch, it should be on the `release-10.4` branch

**Comment by fxthomas** [comment]
Ah, you're right. I'm fairly sure I did it correctly, but hey, that happens. Should be good now.

**Comment by muff1nman** [comment]
No worries - feel free to edit the `release.md` doc if something wasn't clear.

**Pull request merged**